esphome/components/usb_cdc_acm/__init__.py

```diff
@@ -0,0 +1,71 @@
+import esphome.codegen as cg
+from esphome.components import esp32, uart
+from esphome.components.esp32 import add_idf_sdkconfig_option
+from esphome.components.esp32.const import (
+    VARIANT_ESP32P4,
+    VARIANT_ESP32S2,
+    VARIANT_ESP32S3,
+)
+import esphome.config_validation as cv
+from esphome.const import CONF_ID
+
+CODEOWNERS = ["@kbx81"]
+AUTO_LOAD = ["uart"]
+DEPENDENCIES = ["tinyusb"]
+
+CONF_INTERFACES = "interfaces"
+CONF_USB_RX_BUFFER_SIZE = "usb_rx_buffer_size"
+CONF_USB_TX_BUFFER_SIZE = "usb_tx_buffer_size"
+
+usb_cdc_acm_ns = cg.esphome_ns.namespace("usb_cdc_acm")
+USBCDCACMComponent = usb_cdc_acm_ns.class_("USBCDCACMComponent", cg.Component)
+USBCDCACMInstance = usb_cdc_acm_ns.class_("USBCDCACMInstance", uart.UARTComponent)
+
+
+# Schema for individual CDC ACM interface instances
+INTERFACE_SCHEMA = cv.Schema(
+    {
+        cv.GenerateID(): cv.declare_id(USBCDCACMInstance),
+    }
+)
+
+# Main component schema
+CONFIG_SCHEMA = cv.All(
+    cv.Schema(
+        {
+            cv.GenerateID(): cv.declare_id(USBCDCACMComponent),
+            cv.Optional(CONF_USB_RX_BUFFER_SIZE, default=256): cv.uint16_t,
+            cv.Optional(CONF_USB_TX_BUFFER_SIZE, default=256): cv.uint16_t,
+            cv.Optional(CONF_INTERFACES, default=[{}]): cv.All(
+                cv.ensure_list(INTERFACE_SCHEMA),
+                cv.Length(min=1, max=2),  # At least 1, at most 2 interfaces
+            ),
+        }
+    ).extend(cv.COMPONENT_SCHEMA),
+    esp32.only_on_variant(
+        supported=[VARIANT_ESP32P4, VARIANT_ESP32S2, VARIANT_ESP32S3],
+    ),
+)
+
+
+async def to_code(config):
```

Comment:
Try to type the python side of things going forward =)
```suggestion
async def to_code(config: ConfigType) -> None:
```

Reply:
SGTM! 😄